compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -257,13 +258,13 @@ pub fn target_machine_factory(
     };
     let debuginfo_compression = SmallCStr::new(&debuginfo_compression);
 
-    let should_prefer_remapped_for_split_debuginfo_paths =
-        sess.should_prefer_remapped_for_split_debuginfo_paths();
+    let should_prefer_remapped_paths =
+        sess.should_prefer_remapped(RemapPathScopeComponents::DEBUGINFO);
 
     Arc::new(move |config: TargetMachineFactoryConfig| {
         let path_to_cstring_helper = |path: Option<PathBuf>| -> CString {
             let path = path.unwrap_or_default();
-            let path = if should_prefer_remapped_for_split_debuginfo_paths {
+            let path = if should_prefer_remapped_paths {
                 path_mapping.map_prefix(path).0
             } else {
                 path.into()
```

Comment:
This could also make use of the `FilePathMapping::to_real_filename` helper proposed in the other comment.

Reply:
Done.

Btw, this is the place where we cannot have a `Session` and therefor cannot use `for_scope`.